code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,297 @@
+	
+  @geofencing_subscriptions_07_receive_notification_when_device_leaves_geofence
+   Scenario: Receive notification for area-left event
+    Given a valid subscription request body 
+    And the request body property "$.area" is set to circle which covers location "Place1" 
+    And the request body property "$.type" is "area-left" 
+    When the request "createSubscription" is sent
+    Then the response code is 201
+    Then the device left from location "Place1" 
+    Then event notification "area-left" is received on callback-url
+    And sink credentials are received as expected
+    And notification body complies with the OAS schema at "##/components/schemas/EventAreaLeft"
+    And type="org.camaraproject.geofencing-subscriptions.v0.area-left"
+	
+ @geofencing_subscriptions_08_subscriptionExpireTime
+   Scenario: Receive notification for subscription-ends event on expiry  
+    Given a valid subscription request body 
+    And the request body property "$.area" is set to circle which covers location "Place1" 
+    And the request body property "$.type" is "area-left"
+    And the request body property "$.subscriptionExpireTime" set to smaller value
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    Then the subscription is expired
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
+    And the response property "$.terminationReason" is "SUBSCRIPTION_EXPIRED"
+    
+  @geofencing_subscriptions_09_subscriptionMaxEvents
+   Scenario: Receive notification for subscription-ends event on max events reached 
+    Given a valid subscription request body 
+    And the request body property "$.area" is set to circle which covers location "Place1" 
+    And the request body property "$.type" is "area-left"
+    And the request body property "$.subscriptionMaxEvents" is set to 1 
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    Then the device left from location "Place1" 
+    Then event notification "area-left" is received on callback-url
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"And the response property "$.terminationReason" is "MAX_EVENTS_REACHED"
+		
+  @geofencing_subscriptions_10_subscription_delete_event_validation
+   Scenario: Receive notification for subscription-ends event on deletion 
+    Given a valid subscription request body 
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    When the request "deleteGeofencingSubscription" is sent
```

Comment:
First you have to set the path parameter subscriptionId to the value returned by the previous operation